src/main/java/uk/ac/ebi/embl/api/validation/dao/EraproDAOUtilsImpl.java

```diff
@@ -581,6 +581,24 @@ public Entry createMasterEntry(String analysisId, AnalysisType analysisType) thr
 		return masterEntry;
 	}
 
+	public String getCommentsToTranscriptomeMaster(String analysisId) throws SQLException {
+		String analysisQuery = "select " +
+				"XMLSERIALIZE(CONTENT XMLQuery('/ANALYSIS_SET/ANALYSIS/ANALYSIS_TYPE/TRANSCRIPTOME_ASSEMBLY/NAME/text()' PASSING analysis_xml RETURNING CONTENT)) name, " +
```

Comment:
Just a general comment. XmlTable should be more efficient ->
```
         select
            a.name,
            a.platform,
            a.program
         from 
            analysis,
       XMLTABLE('//ANALYSIS_SET/ANALYSIS/ANALYSIS_TYPE/TRANSCRIPTOME_ASSEMBLY'
      PASSING analysis_xml
      COLUMNS 
         name varchar2(4000) PATH 'NAME',
         platform varchar2(4000) PATH 'PLATFORM',
         program varchar2(4000) PATH 'PROGRAM') 
      a
      where analysis_id = ?;
```

However, the main issue is that we are very likely to have to move all XMLs outside of ERAPRO sooner than later and we would need to get into the habit of retrieving the whole XML and parsing it in Java rather than using XmlQuery/Xmltable.

Reply:
Thanks for the suggestion , made changes.